python/dask_cudf/dask_cudf/backends.py

```diff
@@ -10,12 +10,34 @@
 get_parallel_type.register(cudf.Index, lambda _: Index)
 
 
-@meta_nonempty.register((cudf.DataFrame, cudf.Series, cudf.Index))
-def meta_nonempty_cudf(x, index=None):
-    y = meta_nonempty(x.to_pandas())  # TODO: add iloc[:5]
+def _meta_nonempty_simple(x):
+    # TODO: Make sure multiindex types are preserved
+    #       during pandas round-trip...
+    y = meta_nonempty(x.to_pandas())
     return cudf.from_pandas(y)
 
 
+@meta_nonempty.register((cudf.DataFrame, cudf.Series))
+def meta_nonempty_cudf(x, index=None):
+    if isinstance(x.index, cudf.MultiIndex):
+        return _meta_nonempty_simple(x)
+    else:
+        index_dtype = x.index.dtype
+        y = cudf.from_pandas(meta_nonempty(x.to_pandas()))
```

Comment:
I have a fear of round tripping through Pandas, as in many situations Pandas does unnecessary typecasting / type promotion that we do our best not to replicate. Especially in areas for null handling (float64) and lower precision integers (int64).

Reply:
For sure -  I was a bit suprised to see we were still doing this, but I tried to keep my fix "minimal."  I can certainly try to rewrite the metadata creation to avoid using pandas in general.

Reply:
Thanks for the review @kkraus14 - I added some logic to move away from the pandas round-trip for metadata creation.